mlflow/store/file_store.py

```diff
@@ -184,20 +188,20 @@ def get_experiment_by_name(self, name):
         return None
 
     def delete_experiment(self, experiment_id):
-        experiment_dirs = self._get_experiment_path(experiment_id, ViewType.ACTIVE_ONLY)
-        if len(experiment_dirs) == 0:
+        experiment_dir = self._get_experiment_path(experiment_id, ViewType.ACTIVE_ONLY)
+        if experiment_dir is None:
             raise Exception("Could not find experiment with ID %s" % experiment_id)
-        mv(experiment_dirs[0], self.trash_folder)
+        mv(experiment_dir, self.trash_folder)
 
     def restore_experiment(self, experiment_id):
-        experiment_dirs = self._get_experiment_path(experiment_id, ViewType.DELETED_ONLY)
-        if len(experiment_dirs) == 0:
+        experiment_dir = self._get_experiment_path(experiment_id, ViewType.DELETED_ONLY)
+        if experiment_dir is None:
```

Comment:
Raising an exception in `_get_experiment_path` would also allow us to not have to perform this check in the experiment-management methods